### PR TITLE
Add pqsetsinglerowmode to postgres module

### PR DESCRIPTION
**Pull request by artemklevtsov**
Very useful to reduce memory usage when fetch a large amount of data.
Documentation: https://www.postgresql.org/docs/current/libpq-single-row-mode.html

### CODE REVIEW

**Review by dom96 — APPROVED** [review]
LGTM

**Comment by artemklevtsov** [comment]
Also added missing `ExecStatusType` and `ConnStatusType` from the `libpq-fe.h`.

File: lib/wrappers/postgres.nim [hunk 1]
```diff
@@ -248,6 +250,8 @@ proc pqsendQueryPrepared*(conn: PPGconn, stmtName: cstring, nParams: int32,
                           paramLengths, paramFormats: ptr int32,
                           resultFormat: int32): int32{.cdecl, dynlib: dllName,
     importc: "PQsendQueryPrepared".}
+proc pqsetsinglerowmode*(conn: PPGconn): int32{.cdecl, dynlib: dllName,
```
Comment (Araq) [inline lib/wrappers/postgres.nim:253]: This should have been `pqSetSingleRowMode` ...

Comment (timotheecour) [inline lib/wrappers/postgres.nim:253]: the return type should be `cint` (probably other declarations in module are wrong too)

File: lib/wrappers/postgres.nim [hunk 2]
```diff
@@ -248,6 +250,8 @@ proc pqsendQueryPrepared*(conn: PPGconn, stmtName: cstring, nParams: int32,
...
+    importc: "PQsetSingleRowMode".}
```
Comment (timotheecour) [inline lib/wrappers/postgres.nim:254]: add a doc comment with a link; makes searching for docs 1 click away for users.
```nim
## https://www.postgresql.org/docs/current/libpq-single-row-mode.html
```

**Comment by timotheecour** [comment]
@artemklevtsov followup PR welcome to address post review comments

**Comment by timotheecour** [comment]
@Araq I added a label `TODO: followup needed`; if you don't like it just remove or rename it.
I often find it useful to keep track of PR's where a followup is needed, otherwise it's easy to forget.
(the label should be cleared when a followup PR addresses what needs to be addressed)

File: lib/wrappers/postgres.nim (hunk 1 above)
Comment (artemklevtsov) [reply lib/wrappers/postgres.nim:253]: Should I replace all `int32` to `cint` in this module?

Comment (ringabout) [reply lib/wrappers/postgres.nim:253]: fixed

**Pull request merged**